scipy/stats/_continuous_distns.py

```diff
@@ -4192,50 +4195,68 @@ def _entropy(self):
         estimation of the Laplace distribution parameters, so the keyword
         arguments `loc`, `scale`, and `optimizer` are ignored.\n\n""")
     def fit(self, data, *args, **kwds):
-        floc = kwds.pop('floc', None)
-        fscale = kwds.pop('fscale', None)
-
-        _check_fit_input_parameters(data, args,
-                                    kwds, fixed_param=(floc, fscale))
+        data, floc, fscale = _check_fit_input_parameters(self, data,
+                                                         args, kwds)
 
-        # MLE for the laplace distribution
+        # Source: Statistical Distributions, 3rd Edition. Evans, Hastings,
+        # and Peacock (2000), Page 124
 
         if floc is None:
-            loc = np.median(data)
-        else:
-            loc = floc
+            floc = np.median(data)
 
         if fscale is None:
-            scale = (np.sum(np.abs(data - loc))) / len(data)
-        else:
-            scale = fscale
-
-        # Source: Statistical Distributions, 3rd Edition. Evans, Hastings,
-        # and Peacock (2000), Page 124
+            fscale = (np.sum(np.abs(data - floc))) / len(data)
 
-        return loc, scale
+        return floc, fscale
 
 
 laplace = laplace_gen(name='laplace')
 
 
-def _check_fit_input_parameters(data, args, kwds, fixed_param):
-    if len(args) > 0:
-        raise TypeError("Too many arguments.")
-
-    _remove_optimizer_parameters(kwds)
-
-    if None not in fixed_param:
+def _check_fit_input_parameters(self, data, args, kwds):
```

Comment:
```suggestion
def _check_fit_input_parameters(dist, data, args, kwds):
```
If this is not in a class, it shouldn't be `self`